### PR TITLE
feat: add aggregate function min/max support

### PR DESCRIPTION
**Pull request by weijietong**

### CODE REVIEW

File: extensions/functions_arithmetic.yaml [hunk 1]
```diff
@@ -276,3 +276,105 @@ aggregate_functions:
...
+  - name: "min"
+    description: Min a set of values.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
+            required: false
+          - value: i8
+        nullability: DECLARED_OUTPUT
+        decomposable: MANY
+        intermediate: i8
```
Comment (jinfengni) [inline extensions/functions_arithmetic.yaml:288]: Seems both the `intermediate` and `return` type should be `i8?`, indicating that a nullable i8.  

Same applied to other impls.

File: extensions/functions_arithmetic.yaml [hunk 2]
```diff
@@ -276,3 +276,105 @@ aggregate_functions:
         decomposable: MANY
         intermediate: "STRUCT<fp64,i64>"
         return: fp64?
+  - name: "min"
+    description: Min a set of values.
+    impls:
+      - args:
+          - options: [ SILENT, SATURATE, ERROR ]
```
Comment (jacques-n) [inline extensions/functions_arithmetic.yaml:283]: I don't see the purpose of multiple overflow behaviors for min. How would min ever overflow?

File: extensions/functions_arithmetic.yaml (hunk 1 above)
Comment (weijietong) [reply extensions/functions_arithmetic.yaml:288]: Got it. BTW do we need to define another function definication that the `return` type is not nullable ?

Comment (weijietong) [reply extensions/functions_arithmetic.yaml:288]: Got it. BTW do we need to define another function definication that the `return` type is not nullable ? The [specification](https://substrait.io/extensions/) shows that the functoin signature (which is the key in the yaml file) doesn't include the return type.

File: extensions/functions_arithmetic.yaml (hunk 2 above)
Comment (weijietong) [reply extensions/functions_arithmetic.yaml:283]: I just got the meaning of the `options` which indicates the behavior of a function overflow. So will remove it.

File: extensions/functions_arithmetic.yaml (hunk 1 above)
Comment (weijietong) [reply extensions/functions_arithmetic.yaml]: done

File: extensions/functions_arithmetic.yaml (hunk 2 above)
Comment (weijietong) [reply extensions/functions_arithmetic.yaml]: done

**Comment by jacques-n** [comment]
It looks like your commit doesn't match the [expected commit pattern](https://www.conventionalcommits.org/en/v1.0.0-beta.2/). This is a new feature.

**Review by jacques-n — APPROVED** [review]
LGTM. Thanks @weijietong !

**Pull request merged**